proof/append.go

```diff
@@ -160,6 +160,49 @@ func CreateTransitionProof(prevOut wire.OutPoint,
 		TapSiblingPreimage: params.TapscriptSibling,
 	}
 
+	if proof.Asset.IsTransferRoot() {
+		stxoInclusionProofs := make(
+			map[asset.SerializedKey]commitment.Proof,
+			len(proof.Asset.PrevWitnesses),
+		)
+		for _, wit := range proof.Asset.PrevWitnesses {
+			if wit.PrevID == nil {
+				return nil, fmt.Errorf("witness %v has no "+
+					"prevID", wit)
+			}
+
+			prevIdKey := asset.DeriveBurnKey(*wit.PrevID)
```

Comment:
Taking a prevID, then creating a burn key from that, a script key from that, and finally the actual alt leaf is done in many areas in the diff. It may be worth considering the addition of a new helper function to take care of this. It'll cut down on 5-ish lines per instances across the diff.